code/AssetLib/USD/USDLoaderImplTinyusdz.cpp

```diff
@@ -269,8 +268,70 @@ void USDImporterImplTinyusdz::verticesForMesh(
         size_t meshIdx,
         const std::string &nameWExt) {
     UNUSED(nameWExt);
-    pScene->mMeshes[meshIdx]->mNumVertices = static_cast<unsigned int>(render_scene.meshes[meshIdx].points.size());
+    const auto numVertices = static_cast<unsigned int>(render_scene.meshes[meshIdx].points.size());
+    pScene->mMeshes[meshIdx]->mNumVertices = numVertices;
     pScene->mMeshes[meshIdx]->mVertices = new aiVector3D[pScene->mMeshes[meshIdx]->mNumVertices];
+
+    // Check if this is a skinned mesh
+    if (int skeleton_id = render_scene.meshes[meshIdx].skel_id; skeleton_id > -1)
+    {
+        // recursively iterate the joints in the hierarchy
+        std::vector<aiBone *> aiBones;
+        std::vector<const tinyusdz::tydra::SkelNode *> skeletonNodes;
+        skeletonNodes.push_back(&render_scene.skeletons[skeleton_id].root_node);
+        for (int i = 0; i < skeletonNodes.size(); ++i)
+        {
+            const tinyusdz::tydra::SkelNode *skeletonNode = skeletonNodes[i];
+
+            auto outputBone = new aiBone();
+
+            outputBone->mName = aiString(skeletonNode->joint_name);
+            outputBone->mOffsetMatrix = tinyUsdzMat4ToAiMat4(skeletonNode->bind_transform.m).Inverse();
+            aiBones.push_back(outputBone);
+
+            for (const auto &child : skeletonNodes[i]->children)
+            {
+                skeletonNodes.push_back(&child);
+            }
+        }
+
+        const unsigned int numBones = aiBones.size();
+        std::vector<std::vector<aiVertexWeight>> aiBonesVertexWeights;
+        aiBonesVertexWeights.resize(numBones);
+
+        const std::vector<int> &jointIndices = render_scene.meshes[meshIdx].joint_and_weights.jointIndices;
+        const std::vector<float> &jointWeightIndices = render_scene.meshes[meshIdx].joint_and_weights.jointWeights;
+        const int numWeightsPerVertex = render_scene.meshes[meshIdx].joint_and_weights.elementSize;
+
+        const int numJointIndices = jointIndices.size();
```

Comment:
Unused variable `numJointIndices `, will break build if warnings-as-errors enabled

Reply:
Thank you! Fixed!

Reply:
Fixed! Thank you!